### DISCUSSION TITLE
VLS failed to run properly in VS Code

### DISCUSSION THREAD

**Discussion by rust-kotlin** [vlang/v#17661] [Questions and Answers] [answered]
I have installed the VLS using the v ls --install command and set the VLS to open in the configuration of the V extesnion.
But when I open the VS Code, then cd a program, the V extension prompts out that “VLS is not installed. Do you want to install it now?”
After I clicked on the Install button, it soon suggest me that VLS has been installed. But the VLS still doesn't work in my VS Code, should I change some configuratioins?

**Comment by i582** [comment]
> I have installed the VLS using the v ls --install command and set the VLS to open in the configuration of the V extesnion.
> But when I open the VS Code, then cd a program, the V extension prompts out that “VLS is not installed. Do you want to install it now?”
> After I clicked on the Install button, it soon suggest me that VLS has been installed. But the VLS still doesn't work in my VS Code, should I change some configuratioins?

Perhaps this option:
<img width="375" alt="Снимок экрана 2023-03-15 в 11 37 28" src="https://user-images.githubusercontent.com/51853996/225238920-08bbaeb7-6326-423f-b0c3-d8b162554d6d.png">

**Reply by rust-kotlin** [reply]
I had set the VLS to open in the configuration of the V extesnion

**Comment by rust-kotlin** [comment]
Moreover, I am confronted with constant errors when installing the vls, I tried almost 10 times, only once I succeeded in downloading the vls program.(My Internet works fine!)
Here is the detailed information.
```
Spawning v ls --json --check...
[v ls] new data: {"error":{"message":"Language server is not installed nor found.; code: 101","code":"101","details":"- If you are using this for the first time, please run\n  `v ls --install` first to download and install VLS.\n- If you are using a custom version of VLS, check if\n  the specified path exists and is a valid executable.\n- If you have an existing installation of VLS, be sure\n  to remove \"vls.config.json\" and \"bin\" located inside\n  \"%USERPROFILE%\\.vls\" and re-install.\n\n  If none of the options listed have solved your issue,\n  please report it at https://github.com/vlang/v/issues"}}
Spawning v ls --json --update...
[v ls] new data: {"message":"Finding prebuilt executables from GitHub release.."}
[v ls] new data: {"message":"Executable found for this system. Downloading..."}
[v ls] new data: Error 10057 sending data to server (1)
Error performing handshake
{"error":{"message":"response does not start with HTTP/, line: ``","code":"0","details":""}}
```

**Reply by i582** [reply]
Are you using macOS or Windows? On them, VLS is currently unstable, although download problems are something new.

**Reply by rust-kotlin** [reply]
Windows, I will test a little more to try to find out the real problem.
Provide more information about the download problems
```
> Finding prebuilt executables from GitHub release..
> Executable found for this system. Downloading...
Error 10057 sending data to server (1)
Error performing handshake
v ls error: response does not start with HTTP/, line: `` (0)

C:/Users/tom/AppData/Local/Temp/v_0/vls.11772736471763394746.tmp.c:9226: at print_backtrace: Backtrace
C:/Users/tom/AppData/Local/Temp/v_0/vls.11772736471763394746.tmp.c:20958: by main__VlsUpdater_cli_error
C:/Users/tom/AppData/Local/Temp/v_0/vls.11772736471763394746.tmp.c:21065: by main__main
C:/Users/tom/AppData/Local/Temp/v_0/vls.11772736471763394746.tmp.c:21527: by wmain
004c7448 : by ???
004c75ab : by ???
7fff06df26bd : by ???
```

**Comment by rust-kotlin** [comment, marked as answer]
Somehow, mannually specify the binary path of the vls works for me. But the download problem truly exists.
